### ISSUE TITLE
Discover Kafka cluster members automatically

### ISSUE THREAD

**Issue by jfallows** [reaktivity/nukleus-kafka.java#1]
Kafka protocol supports deployment as a cluster of brokers, solving for broker discovery via the `MetaData` API which returns a list of brokers with distinct IP addresses and port numbers.

Therefore, the routes to these broker endpoints are quite dynamic.

Using reaktivity/nukleus-tcp.java#51, we can reuse the same `targetRef` on `tcp` nukleus but include `TcpBeginEx` to guide the connection to the correct broker.

```
Metadata Response (Version: 1) => [brokers] controller_id [topic_metadata] 
  brokers => node_id host port rack 
    node_id => INT32
    host => STRING
    port => INT32
    rack => NULLABLE_STRING
  controller_id => INT32
  topic_metadata => topic_error_code topic is_internal [partition_metadata] 
    topic_error_code => INT16
    topic => STRING
    is_internal => BOOLEAN
    partition_metadata => partition_error_code partition_id leader [replicas] [isr] 
      partition_error_code => INT16
      partition_id => INT32
      leader => INT32
      replicas => INT32
      isr => INT32
```

The `MetaData` response identifies the leader and replica brokers for each topic partition, which we must honor to ensure we are sending `Fetch` requests to the correct broker for a given topic partition.

Given that cluster members can come and go, the leader and replicas for a given topic partition can change over time, which is most easily detected by [error codes](https://kafka.apache.org/0100/protocol.html#protocol_error_codes) such as `UNKNOWN_TOPIC_OR_PARTITION`.

**Comment by cmebarrow** [comment]
Useful links:
- [Kafka protocol guide](http://kafka.apache.org/protocol.html): this is the official documentation
- [A Guide To The Kafka Protocol](https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol): Defines message set and record batch, maybe more up-to-date and detailed

**Comment by cmebarrow** [comment]
Spec scripts for metadata should include:
- request all topics (null topic array on request)
- request specific topics
- response includes multiple topics
- response includes a topic with multiple partitions on same node
- response includes a topic with multiple partitions across different nodes
- response includes a topic with an error code
- response includes a partition with an error code

**Comment by jfallows** [comment]
Fixed by #3
